### DISCUSSION TITLE
Many "Invalid session" errors

### DISCUSSION THREAD

**Discussion by rossmacarthur** [miguelgrinberg/python-socketio#1322] [Q&A]
We are running socket.io with a TypeScript frontend and `python-socketio` on the backend. We often get logs like

```
Invalid session LThYhtODB-Y8mnlbAACm (further occurrences of this error will be logged with level INFO)
```

Is this really an error? In what contexts does this happen? Is the backend or frontend doing something wrong? Nothing appears to be wrong in normal operation and the socket.io connection works fine in our product.

**Comment by miguelgrinberg** [comment]
If this happens constantly then I think it should be investigated.

It means that your client and your server do not recognize each other. This could happen, for example, if you have multiple servers in front of a load balancer, and for some reason the load balancer decides to reassign a given client to a different server worker that has no previous knowledge of the client.

**Comment by muthupriyashanmugam** [comment]
@rossmacarthur  Did you able to fix those errors on your side? if so, can you share me the points to resolve it. Because i am facing the same issue now.

**Reply by rossmacarthur** [reply]
No, I still have this issue. We don't use multiple socket.io servers so it wasn't related to load balancing as suggested in the comment above. However I suspect the issue is actually on the client side, but was never able to figure it out. We are using `socket.io-client` TypeScript package on the frontend.

**Comment by francipvb** [comment]
Hello,

We have this issue here. I am not sure how to configure my server (dokku EC2 instance) to assign the client to the same server.

**Comment by aykutaaykut** [comment]
I am getting the same problem and using fast-api in the backend along with socketio server together + `socket.io-client` package in the frontend. Did you find the root reason or any fix you find already?

**Reply by rossmacarthur** [reply]
We did manage to fix this but I can't remember which of the following was the fix. I looked in our git history and we have these two commits:

We set the transports on the client to prefer `websocket`
```
 transports: ['websocket', 'polling'],
```

On the server side we increased the ping timeout to 60s

```
socketio.AsyncServer(
    ...,
    ping_timeout=60,
)
```

**Reply by aykutaaykut** [reply]
thank you very much, I will try both
